### PR TITLE
Add support for relative $schema paths

### PR DESCRIPTION
**Pull request by nspire909**
I also fixed an issue with the tests. The filename being passed in to eslint by the tests was only a partial path in most circumstances, and this fix needs `context.getFilename()` to return a full path. I'm not very familiar with eslint, so if there is a better way to do this, let me know.

### CODE REVIEW

File: src/rules/no-invalid.ts [hunk 1]
```diff
@@ -448,7 +449,16 @@ export default createRule("no-invalid", {
                 const program = node as TOML.TOMLProgram
                 $schema = findSchemaPathFromTOML(program)
             }
-            return typeof $schema === "string" ? $schema : null
+            return typeof $schema === "string"
+                ? $schema.startsWith(".")
+                    ? path.resolve(
+                        path.dirname(
+                            getPhysicalFilename(context.getFilename()),
```
Comment (JounQin) [inline src/rules/no-invalid.ts:456]: Should prefer `context.getPhysicalFilename()` if it's available.

**Review by JounQin — CHANGES_REQUESTED** [review]

File: src/rules/no-invalid.ts (hunk 1 above)
Comment (JounQin) [reply src/rules/no-invalid.ts:456]:
```suggestion
                            typeof context.getPhysicalFilename === 'function' ? context.getPhysicalFilename() : getPhysicalFilename(context.getFilename()),
```

**Review by ota-meshi — APPROVED** [review]
LGTM! Thank you!

**Review by JounQin — APPROVED** [review]
LGTM

**Pull request merged**